src/crimson/tools/store_bench/store-bench.cc

```diff
@@ -0,0 +1,685 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:nil -*-
+
+/**
+ * crimson-store-bench
+ *
+ * This tool measures various IO patterns against the crimson FuturizedStore
+ * interface.
+ *
+ * Usage should be:
+ *
+ * crimson-store-bench --store-path <path>
+ *
+ * where <path> is a directory containing a file block.  block should either
+ * be a symlink to an actual block device or a file truncated to an appropriate
+ * size if performance isn't relevant (testing or developement of this utility,
+ * for instance).
+ *
+ * One might want to add something like the following to one's .bashrc to
+ * quickly run this utility during development from build/:
+ *
+ */
+
+#include <random>
+#include <vector>
+
+#include <boost/program_options/parsers.hpp>
+#include <boost/program_options/variables_map.hpp>
+
+#include <seastar/apps/lib/stop_signal.hh>
+#include <seastar/core/app-template.hh>
+#include <seastar/core/byteorder.hh>
+#include <seastar/core/future-util.hh>
+#include <seastar/core/gate.hh>
+#include <seastar/core/reactor.hh>
+#include <seastar/core/rwlock.hh>
+#include <seastar/core/thread.hh>
+#include <seastar/util/defer.hh>
+
+#include "crimson/common/config_proxy.h"
+#include "crimson/common/coroutine.h"
+#include "crimson/common/log.h"
+
+#include "crimson/os/futurized_collection.h"
+#include "crimson/os/futurized_store.h"
+
+namespace po = boost::program_options;
+
+using namespace ceph;
+
+SET_SUBSYS(osd);
+
+/**
+ *These are functions that are used in both types of work_loads
+ */
+
+/**
+ * Creates a unique object
+ */
+ghobject_t create_hobj(unsigned id) {
+  return ghobject_t(
+      shard_id_t::NO_SHARD, seastar::this_shard_id(),
+      id, // hash, normally rjenkins of name, but let's just set it to id
+      "", // namespace, empty here
+      "", // name, empty here
+      0,  // snapshot
+      ghobject_t::NO_GEN);
+};
+
+/**
+ * This function creates a collection(PG) with pool id as obj_id
+ * The function is called when we create an object
+ * the goal is to have each object belong to a separate collection
```

Comment:
Each object being in a distinct collection is specific to the pg log workload (each pg has a log and a collection).  The index workload should have an argument specifying the number of objects per collection.